tests/unit/test_query_language.py

```diff
@@ -145,6 +145,14 @@ def test_dataset_rejects_invalid_variable_names(variable_name, error):
         setattr(Dataset(), variable_name, patients.i)
 
 
+def test_cannot_reassign_population():
+    dataset = Dataset()
+    dataset.define_population(patients.exists_for_patient())
+    dataset.year_of_birth = patients.date_of_birth.year
+    with pytest.raises(AttributeError, match="already defined"):
+        dataset.define_population(dataset.year_of_birth <= 2000)
```

Comment:
Why are we using a different population condition?

Reply:
true, it should be simpler because the difference isn't relevant and we're not testing it.